src/snapred/backend/recipe/DiffractionCalibrationRecipe.py

```diff
@@ -95,11 +107,15 @@ def executeRecipe(self, ingredients: Ingredients, groceries: Dict[str, str]) ->
         groupedAlgo.setProperty("GroupingWorkspace", self.groupingWS)
         groupedAlgo.setProperty("Ingredients", ingredients.json())
         groupedAlgo.setProperty("PreviousCalibrationTable", self.calTable)
+        groupedAlgo.setProperty("MaskWorkspace", self.maskWS)
         groupedAlgo.setProperty("FinalCalibrationTable", self.calTable)
         try:
             groupedAlgo.execute()
+            # TODO: these output workspaces are *optional*: why are they included here without checking?
+            #   And, how to check if they're still default values?
```

Comment:
If you make them mandatory, then the mantid API will (errantly) demand they already exist in the ADS.  The way to ensure they exist is to add an additional method,
``` python
def validateInputs(self) -> Dict[str, str]:
    errors = {}
    if self.getProperty("OutputWorkspace").isDefault:
        errors["OutputWorkspace"] = "error message here"
    return errors
```
This will be called as part of the mantid algorithm lifecycle and will throw a runtime error including the key and the message.  Later algorithms in SNAPRed have examples of using this.  See for instance [`FetchGroceriesAlgorithm`](https://github.com/neutrons/SNAPRed/blob/04d31430dcd422cd164a4c0cde0e70515bae5e7c/src/snapred/backend/recipe/algorithm/FetchGroceriesAlgorithm.py#L75C1-L85C22).

Reply:
This is an interesting issue.  `Output` parameters _may_ exist, or may not.  The Mantid documentation does mention that an `Output` workspace which already exists should be used (i.e. as "input") by an algorithm.  I believe that this is an alternative, and also correct way that this issue can be dealt with.  That is, as an _optional_ `Output` parameter.  This is what we did for the corresponding masking changes on the Mantid side (e.g. in `GetDetectorOffsets` and `PDCalibration`).  If the user has not specified the parameter, it is deleted at the end of execution.